test/e2e/gcp_test.go

```diff
@@ -0,0 +1,207 @@
+// +build e2e
+
+package e2e
```

Comment:
Why making this file `*_test.go` file? It doesn't have a single go test.
Same question for the aws and azure provider.

Reply:
1. we have to get the access of global variable which are test scope  ==> since as part of previous commit it is clear this need not required.
2.  we are planning to introduce more test cases specific to provider so , provider specific test will fall here.  ==>

as of not moving back with out  **_test**.